### ISSUE TITLE
Typing for the pizzly-node client

### ISSUE THREAD

**Issue by Nightbr** [Bearer/Pizzly#220]
Hey! 

I have integrated the pizzly-node in application but it seems there is no typing for TypeScript. 

`import Pizzly from 'pizzly-node';` -> `Pizzly` is type `any`

Seems it is the bundling options of the library which will not create an ESM module and have the typing for TypeScript.

**Workaround**: redifined an interface and type with the interface:

```typescript
import { ConfigService } from '@nestjs/config';
import Pizzly from 'pizzly-node';

// @TODO Typing for the Pizzly service
interface OAuthService {
  integration(name: string): OAuthService;
  auth(id: string): OAuthService;
  get(path: string, queryParams): Response;
}

export const oauthServiceFactory = {
  provide: 'OAUTH_SERVICE',
  useFactory: (configService: ConfigService): OAuthService => {
    return new Pizzly(configService.get('oauthService')) as OAuthService;
  },
  inject: [ConfigService],
};
```

The pizzly-node client is written in TypeScript so I think it's just a config of the bundler to package with typings.

Any idea on this?

EDIT:

See [this article](https://blog.logrocket.com/publishing-node-modules-typescript-es-modules/)

We should configure the tsconfig a bit better https://github.com/Bearer/Pizzly/blob/master/src/clients/node/tsconfig.json

Publish a commonJs package for legacy but also an esm package for modern apps.

**Comment by markmichon** [comment]
Definitely seems achievable. I'll see if I can get this working.

**Comment by Nightbr** [comment]
Seems just enabling declaration should work:

```
{
  "compilerOptions": {
    "strict": true,
    "declaration": true, // generates declaration files
    "esModuleInterop": true
  }
}
```

**Comment by Nightbr** [comment]
Closed by #226
